realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -720,7 +723,12 @@ public static void insert(Realm realm, Iterator<? extends RealmModel> objects, M
         while (objects.hasNext()) {
             object = (some.test.AllTypes) objects.next();
             if(!cache.containsKey(object)) {
-                long rowIndex = Table.nativeAddEmptyRow(tableNativePtr, 1);
+                long rowIndex;
+                if (object instanceof RealmObjectProxy && ((RealmObjectProxy)object).realmGet$proxyState().getRealm$realm() != null && ((RealmObjectProxy)object).realmGet$proxyState().getRealm$realm().getPath().equals(realm.getPath())) {
+                    rowIndex = ((RealmObjectProxy)object).realmGet$proxyState().getRow$realm().getIndex();
```

Comment:
If it is managed, then it will never be really used again i think? just a `continue` makes the generator logic a bit simpler, save one entry in map and not harmful at all. I will just add a continue here.